FreeRTOS_IP.c

```diff
@@ -1171,8 +1171,18 @@ BaseType_t FreeRTOS_IPInit( const uint8_t ucIPAddress[ ipIP_ADDRESS_LENGTH_BYTES
     configASSERT( sizeof( UDPHeader_t ) == ipEXPECTED_UDPHeader_t_SIZE );
 
     /* Attempt to create the queue used to communicate with the IP task. */
-    xNetworkEventQueue = xQueueCreate( ipconfigEVENT_QUEUE_LENGTH, sizeof( IPStackEvent_t ) );
-    configASSERT( xNetworkEventQueue != NULL );
+    #if( configSUPPORT_STATIC_ALLOCATION == 1 )
+    {
+        static StaticQueue_t xNetworkEventStaticQueue;
+        uint8_t ucNetworkEventQueueStorageArea[ipconfigEVENT_QUEUE_LENGTH * sizeof(IPStackEvent_t)];
+        xNetworkEventQueue = xQueueCreateStatic(ipconfigEVENT_QUEUE_LENGTH, sizeof(IPStackEvent_t), ucNetworkEventQueueStorageArea, &xNetworkEventStaticQueue);
```

Comment:
It looks like it needs some formatting by Uncrustify, but the changes all look good.